### ISSUE TITLE
Cannot pass function as an argument

### ISSUE THREAD

**Issue by taless474** [STEllAR-GROUP/phylanx#904]
```py
import numpy as np
from phylanx import Phylanx, PhylanxSession, execution_tree

@Phylanx
def fn():
	return lambda a, b : 2 * a - b
	
@Phylanx
def foldl_eager(fn, elems, initializer):
	return fold_left(fn, initializer, elems)
	
foldl_eager(fn, [1, 2, 3], 3)
```
## Expected
```pytb
13
```
## Actual
```pytb
Traceback (most recent call last):
  File "test32.py", line 15, in <module>
    foldl_eager(fn, [1, 2, 3], 3)
  File "C:\Repos\phylanx\cmake-build-debug\python\build\lib.win-amd64-3.6\phylanx\ast\transducer.py", line 94, in __call__
    result = self.backend.call(args)
  File "C:\Repos\phylanx\cmake-build-debug\python\build\lib.win-amd64-3.6\phylanx\ast\physl.py", line 483, in call
    return self.lazy(args).eval()
  File "C:\Repos\phylanx\cmake-build-debug\python\build\lib.win-amd64-3.6\phylanx\ast\physl.py", line 450, in eval
    *self.args)
RuntimeError: Unable to cast Python instance of type <class 'phylanx.ast.transducer.Phylanx.<locals>.__PhylanxDecorator'> to C++ type 'phylanx::execution_tree::primitive_argument_type'
```

**Comment by hkaiser** [comment]
This is an interesting one! @rtohid Would you see a way to identify decorated functions and to handle things properly when one of those is being passed to another decorator?

**Comment by hkaiser** [comment]
@taless474 this should be fixed now, please see #917
